pom.xml

```diff
@@ -68,7 +68,7 @@
         <dependency>
             <groupId>org.javassist</groupId>
             <artifactId>javassist</artifactId>
-            <version>3.24.0-GA</version>
+            <version>3.23.1-GA</version>
```

Comment:
Uh... I missed that out :\

Reply:
Hello @lukaszlenart .

The only reason I noticed it was I ran the Maven build with JDK7 (and it failed due to "Unsupported major.minor version 52.0" related to Javassist).   :)

It seems like they jumped from source/target 1.6 right to 1.8 in 3.24.0-GA (the commit that made the change didn't explain why ...)

Reply:
Thanks :)

Reply:
Does OGNL even need javassist anymore? I'm not sure if anyone ever used those features other than me in Tapestry 4. It might be safe to just drop that feature set instead. Up to you. =)

Reply:
Hello @jkuhnert .

Not sure if anyone out there needs `Javassist` for `OGNL`.  If neither you nor @lukaszlenart are aware of any active Javassist feature usage, maybe it could marked for deprecation in the `OGNL 3.2.x` line (_or simply phased out in future 3.2.x and noted in the release notes_) ?

I was just trying to keep it compiling on JDK 7.  :)

Reply:
It's used in `ExpressionCompiler` and I don't know if this can be easily replaced

Reply:
Hello @lukaszlenart .

Based on your comment about `ExpressionCompiler` and a brief look at the code, it certainly looks like Javassist is pretty fundamental to its design... 

As a "heads-up" the Javassist team accepted a PR (https://github.com/jboss-javassist/javassist/pull/230) to restore JDK 7 runtime compatibility.  That could mean that the next `Javassist` `3.24.x` or `3.25` release might work for the `OGNL 3.2.x` line, whenever the next Javassist official build is released.

Reply:
Great news, thanks a lot!!!